Code/autopkglib/DmgCreator.py

```diff
@@ -143,16 +143,15 @@ def main(self):
 
         # Call hdiutil.
         try:
-            proc = subprocess.Popen(
-                cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True
-            )
-            (_, stderr) = proc.communicate()
+            proc = subprocess.run(cmd, capture_output=True, text=True)
         except OSError as err:
             raise ProcessorError(
                 f"hdiutil execution failed with error code {err.errno}: {err.strerror}"
             )
         if proc.returncode != 0:
-            raise ProcessorError(f"creation of {self.env['dmg_path']} failed: {stderr}")
+            raise ProcessorError(
+                f"creation of {self.env['dmg_path']} failed: {proc.stderr}"
+            )
```

Comment:
Same feedback here as above - we should leverage `check=True`.

Honestly, we use this construction enough times that I'm starting to think we should make a helper function for it in autopkglib, but not something that needs to be in this PR.

Reply:
cfe0b9d